api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -73,7 +75,10 @@ class OAuth2Client implements OAuth2Authenticator, Closeable {
         config.getGrantType().requiresUserInteraction()
             ? CompletableFuture.allOf(started, used)
             : started;
-    currentTokensStage = ready.thenApplyAsync((v) -> fetchNewTokens(), executor);
+    currentTokensStage =
+        ready
+            .thenApplyAsync((v) -> fetchNewTokens(), executor)
+            .thenApply(this::maybeExchangeTokens);
```

Comment:
Do we need `thenApplyAsync` here (and below)? I wonder, because the call to `fetchNewTokens` is async.

Reply:
Async is not needed: the code will be executed by whatever thread executed the previous step, so here: an executor thread, which is what we want.